pegasus_wrapper/workflow.py

```diff
@@ -212,7 +275,10 @@ def _run_python_job(
         job_is_stageable: bool = False,
         job_bypass_staging: bool = False,
         times_to_retry_job: int = 0,
+        job_profiles: Optional[List[PegasusProfile]] = None,
```

Comment:
Would it be possible to default this to an empty list? Then you could do away with some checks like `if job_profiles`